.github/workflows/reusable_trivy-scans.yml

```diff
@@ -1,51 +1,39 @@
 ---
 name: Run Trivy Scans
+
 on:
-    workflow_call:
-jobs:
-    build-image:
-        runs-on: ubuntu-latest
-        steps:
-            - name: Check out GitHub Repo
-              uses: actions/checkout@v4
+  workflow_call:
 
-            - name: Build an image from Dockerfile
-              run: |
-                  git config --global --add safe.directory $GITHUB_WORKSPACE;
-                  docker build -t trivy-test .
+jobs:
+  build-and-scan-image:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      security-events: write
 
-            - name: Run Trivy vulnerability scanner, output as table
-              uses: aquasecurity/trivy-action@master
-              with:
-                  image-ref: "trivy-test"
-                  format: "table"
-                  output: "trivy-results.tbl"
-                  timeout: "20m0s"
+    steps:
+      - name: Check out GitHub Repo
+        uses: actions/checkout@v4
 
-            - name: Check for log4j CVEs
-              run: |
-                  set -e
-                  for cve in CVE-2021-4104 CVE-2021-44228 CVE-2021-45046 CVE-2022-22965 ; do
-                  	if grep -c $cve trivy-results.tbl > 0;
-                  	  then
-                  	    echo "Found log4j error: $cve"
-                  	    exit 2
-                  	fi
-                  done
-                  # If above doesn't find errors, return pass message:
-                  echo "Did not find core log4j errors, yay!"
+      - name: Build an image from Dockerfile
+        run: |
+          docker build -t kbase/trivy:scan . 
 
-            - name: Rerun Trivy vulnerability scanner for GitHub security
-              uses: aquasecurity/trivy-action@master
-              if: always()
-              with:
-                  image-ref: "trivy-test"
-                  format: "sarif"
-                  output: "trivy-results.sarif"
-                  timeout: "20m0s"
-                  ignore-unfixed: true
```

Comment:
I'm confused - shouldn't this have resulted in the vulns being uploaded?

Reply:
Good question. I'm not sure why. It could be that uploads are not supported when called in this manner, and they need to explicitly grant permissions to uploading within the action. Or its a github bug? I've already tested it by
* forking .github
* forking service_wizard2 
and the security tab is not being populated. 

 If this is the case, then we need to go and add a trivy workflow manually to each repo, such as I am doing here https://raw.githubusercontent.com/kbase/service_wizard2/refs/heads/main/.github/workflows/trivy.yaml

Reply:
Yea this is unfortunate.. I can revert these changes and instead add a new workflow and we will have to go an issue a PR adding trivy to each repo.

Reply:
Ok, so sounds like this needs more investigation IIUC

Reply:
Ugh, cross post. You're sure there's no way to get it to work without updating each repo?

Reply:
fair enough

Reply:
The logs say it is being uploaded, but then I don't see anything on the github security tab

Reply:
Hmm, maybe GH support is the next step them

Reply:
I made a support ticket.